src/backend/marsha/core/api/video.py

```diff
@@ -686,3 +686,56 @@ def end_sharing(
         serializer = self.get_serializer(video)
         channel_layers_utils.dispatch_video_to_groups(video)
         return Response(serializer.data)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="participants",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def participants(
+        self,
+        request,
+        pk=None,
+    ):
+        """Stores participants list in a live stream.
+
+        Broadcasts a xmpp message to all users in the video room.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint, it should contain a payload with data to be stored.
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+
+        live_participants_serializer = serializers.LiveParticipantsSerializer(
+            data=request.data
+        )
+        if not live_participants_serializer.is_valid():
+            return Response(
+                live_participants_serializer.errors, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        video = self.get_object()
+        video.participants_asking_to_join = request.data.get(
+            "participants_asking_to_join"
+        )
+        video.participants_in_discussion = request.data.get(
+            "participants_in_discussion"
+        )
```

Comment:
You should use the validated data from the serializers instead using the one from the request.